### ISSUE TITLE
No visual diffeence of the piano-roll-grid in 1/4, 1/2 & 1/1

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#5363]
### Bug Summary
LMMS has same grid in 1/4, 1/2 & 1/1

#### Steps to reproduce
look..

#### Expected behavior
There ought to be a change in grid, when Q is changed 

#### Actual behavior
Theres non

#### Screenshot
![image](https://user-images.githubusercontent.com/6662066/71693417-b73bdd00-2dac-11ea-8354-d2f2f7aa3cd1.png)

#### LMMS version used
1.2.1 on win10, but this is ofcause all versions
#### Logs'
non

**Comment by Veratil** [comment]
This is because the dark green lines are drawn to show time signature beats per measure. I'm not sure how to fix this without removing those lines.

Do you have an idea of how to handle this @musikBear?

**Comment by Spekular** [comment]
I don't think this is an issue, personally. We don't draw 192 lines for 1/192 at default zoom because that'd be too dense, and I believe there's also such a thing as too sparse.

**Comment by musikBear** [comment]
@Veratil not without removing the green lines .. Perhaps replace them with lines of a different douched color..
It just seams strange, that a change in the ui of Q *apparently* does noting, (and it is different in f.i. pro-tool i dont know how this is handled in fl or the others, though, but i managed to get confused, and was quite surprised that it was like that.. I remember it different(?) )

**Comment by musikBear** [comment]
@Spekular we do draw grey lines differently 
The 'jump' from 1/48 to 1/192 creates very different grids

**Comment by JohannesLorenz** [comment]
> It just seams strange, that a change in the ui of Q apparently does noting

I think you mean "nothing"? It *does* do something: With "1/4", you can add a note at the 2nd 4th. With "1/2", you cannot.

Personally, I agree with @Spekular : This is not a bug (changing to enhancement). And I also like the green lines at 1/4, personally.

More opinions?

**Comment by zonkmachine** [comment]
@musikBear If I select quantization 1/1 and a time signature of 5/4. Where do you want the grid lines?

**Comment by Veratil** [comment]
> @musikBear If I select quantization 1/1 and a time signature of 5/4. Where do you want the grid lines?

A whole note is still a whole note, and as such I believe the alignment should be at the start of a bar. Time signature represents the bar, and we're currently drawing for the "whole song". With odd time signatures this really starts to show itself in the grid because we're trying to divide up things unevenly, starting from bar 1 instead of the current bar.

So if we have 5/4, this is what 1/1 quantization would represent:
```
bar:  +         +         +
beat: | | | | | | | | | | |
quan: |       | |       |
```

I think this type of representation would fit for any type of odd time signature, because we're considering the quantization of the bar instead of the entire song from bar 1. I understand we're tied to MIDI here as well using `1/192` intervals, so at that point we really need to figure out how to tie it all together, especially with non-powers-of-two denominators, such as 5, 7, etc.

Now with that said, with *how* to draw the grid we'll want to have a perceived layer order. Currently the order of layers drawn in the Piano Roll are (from furthest behind to front):
1. Quantization lines
2. *(not part of lines) Piano roll + horizontal key lines*
3. *(not part of lines) Alternating bar shading*
4. Beat lines
5. Bar lines
6. *(and on) rest of the piano roll layers*

Because of this ordering, the quantization lines are drawn over by the bar/beat lines. If we change this order then the quantization lines becomes a little more complicated to draw (since we'd be drawing over beat/bar lines).

I think I've gone a little off topic here, but I hope that gives some food for thought. 😁

**Comment by zonkmachine** [comment]
Borrowing your ascii art a bit. This is how notes are quantized today with 5/4 beat.

```
bar:  +         +         +
beat: | | | | | | | | | | |
quan: |       |       |
```

...or we can just stick with the 1/4 quantization grid?   :)

**Comment by Veratil** [comment]
> ...or we can just stick with the 1/4 quantization grid? :)

Remove the 1/1, 1/2, and 1/4, eh? I like your style. 😂

But this is a problem that goes deeper than the no visual difference, which I don't want to off topic this issue to, but there needs to be a rethinking of the way it's displayed overall.

**Comment by Veratil** [comment]
So playing around with the beat lines, how's this idea? Didn't include the bar lines though, and this is just poorly coded in test.
![lmms-quan-11-43](https://user-images.githubusercontent.com/2244945/80546272-d9202780-897a-11ea-89c3-3ce936aa802e.PNG)
![lmms-quan-11-44](https://user-images.githubusercontent.com/2244945/80546273-dae9eb00-897a-11ea-9375-cf906642f042.PNG)
![lmms-quan-12-43](https://user-images.githubusercontent.com/2244945/80546279-dcb3ae80-897a-11ea-8b58-68852f518e56.PNG)
![lmms-quan-12-44](https://user-images.githubusercontent.com/2244945/80546281-dde4db80-897a-11ea-9977-7b5afa340dd7.PNG)
![lmms-quan-14-43](https://user-images.githubusercontent.com/2244945/80546287-dfae9f00-897a-11ea-913d-e3d666fb81c1.PNG)
![lmms-quan-14-44](https://user-images.githubusercontent.com/2244945/80546292-e1786280-897a-11ea-82ea-11c9c625d358.PNG)
![lmms-quan-13-43](https://user-images.githubusercontent.com/2244945/80546450-48961700-897b-11ea-8531-ae50a309a210.PNG)
![lmms-quan-13-44](https://user-images.githubusercontent.com/2244945/80546454-4a5fda80-897b-11ea-8180-74805bf5c544.PNG)
![lmms-quan-16-43](https://user-images.githubusercontent.com/2244945/80546466-4d5acb00-897b-11ea-943c-3e9fce626774.PNG)
![lmms-quan-16-44](https://user-images.githubusercontent.com/2244945/80546461-4b910780-897b-11ea-8f43-78d9acf68f0a.PNG)

**Comment by musikBear** [comment]
@Veratil  Great that you addresses this
What do you think about 
![image](https://user-images.githubusercontent.com/6662066/80738477-6303f800-8b15-11ea-8000-19281ba4a4be.png)
When Q has the red settings, there are no option for placing notes in between the vertical dividers, so i think this would be fully logical in respect to what we have for 1/8 and smaller.
The second part of the dividers starting with 1/3 i feel we should rethink
In the code, there is no doubt of what they are: They are triplets!
Why not *name* them triplets.
Users often ask how to make triplets, and then we have to tell them, that lmms already has triplets, they are just named different(?)
(I am fully aware of the fact that the symbol in the note-select shows the triplet symbol, but few users knows this note-nomenclature 
![image](https://user-images.githubusercontent.com/6662066/80739371-a01cba00-8b16-11ea-8e7a-9740519e61aa.png)
but it is definitively already there! It only a question if it is obvious and clear to everyone.
